File: challenges/calculate-mode/js/calculateMode.js
const calculateMode = (array) => {
    const mode = []
    const counts = {}
    let max = 0

    for (let i of array) { // track each count and max count
        counts[i] = isNaN(counts[i]) ? 1 : counts[i] += 1
        max = counts[i] > max ? counts[i] : max
    }

    for (let k of Object.keys(counts)) { // add modes to list
        if (counts[k] === max) {
            if (isNaN(Number(k))) {
                mode.push(k)
            } else {
                mode.push(Number(k))
            }
        }
    }

    return mode
}

module.exports = { calculateMode }

File: challenges/calculate-mode/python/calculate_mode.py
from collections import defaultdict


def calculate_mode(array):
    mode = list()
    counts = defaultdict(int)
    max = 0

    for i in array:  # track each count and max count
        counts[i] += 1
        max = counts[i] if counts[i] > max else max

    for k, v in counts.items():  # add modes to list
        if v == max:
            mode.append(k)

    return mode
